app/models/user.rb

```diff
@@ -52,6 +52,12 @@ def following?(other_user)
     following.include?(other_user)
   end
 
+  def self.filter_users(filter)
+    User.where('LOWER(nickname) LIKE ?',
+               "%#{filter}%").or(User.where('LOWER(first_name) LIKE ?',
+                                            "%#{filter}%")).or(User.where('LOWER(last_name) LIKE ?', "%#{filter}%"))
```

Comment:
My bad, I meant filtering by full name (so if you type 'Cande Lopez' you get the user, in this scenario we won't get the user as there would be no user with first name 'Cande Lopez' or last name 'Cande Lopez').
You should be able to do this using SQL's CONCAT function